src/util/system.cpp

```diff
@@ -506,14 +506,46 @@ std::string ArgsManager::GetHelpMessage() const
     return usage;
 }
 
+std::string ArgsManager::GetSpecificHelpMessage()
+{
+    const std::string helpParams[] = {"-?", "-h", "-help", "-help-debug"};
+    std::string helpName = "";
+    for (long unsigned int i = 0; i < sizeof(helpParams); ++i) {
+        if (this->IsArgSet(helpParams[i])) {
+            helpName = this->GetArg(helpParams[i], "");
+            break;
+        }
+    }
+    // Return unspecific help if no arg was given
+    if (helpName == "")
+        return this->GetHelpMessage();
+
+    std::pair<std::string, ArgsManager::Arg> command;
+    bool cmdExists = false;
+    LOCK(cs_args);
+    for (const auto& arg_map : m_available_args) {
```

Comment:
Please factor this out to a function to get info for a specified setting name instead of having this loop in-line.

Reply:
What's the point of porting this to a function. It is only used once.

Reply:
It helps encapsulate logic by naming a block of code. I don't have strong opinions on this but I think it would be helpful for readers. It looks like this loop is determining if the provided `helpName` is a valid command name. By naming this block of code it becomes significantly easier to reason about IMO.